src/domain/task-system/services/src/task_executor_impl.rs

```diff
@@ -99,9 +102,34 @@ impl TaskExecutor for TaskExecutorImpl {
 
             let outcome = match &task.logical_plan {
                 LogicalPlan::UpdateDataset(upd) => {
+                    let dataset_env_vars = DatabaseTransactionRunner::new(self.catalog.clone())
+                        .transactional_with(
+                            |dataset_env_vars_svc: Arc<dyn DatasetEnvVarService>| async move {
+                                let dataset_env_vars = dataset_env_vars_svc
+                                    .get_all_dataset_env_vars_by_dataset_id(
```

Comment:
This code is growing too big, let's split handlers for each type of logical plan into separate methods for the beginning.